### PR TITLE
Configure the platform for DTC support

### PR DESCRIPTION
**Pull request by mjg59**
Enable the necessary Kubernetes functionality for DTC, and populate it with the
appropriate API object types.

### CODE REVIEW

**Comment by dghubble** [comment]
Can you rebase on #274? There are a few conflicts to [resolve](Tested after rebase at https://github.com/dghubble/coreos-baremetal/tree/dtc).

Are there there followup docs or links that should go with this? To answer user questions "what do I get by enabling this?" and "how do I verify its working?".

File: examples/ignition/k8s-master.yaml [hunk 1]
```diff
@@ -161,11 +161,18 @@ storage:
               - --service-cluster-ip-range={{.k8s_service_ip_range}}
               - --secure-port=443
               - --advertise-address={{.ipv4_address}}
+              {{ if index . "enable_dtc" }}
+              - --admission-control=TaintAdmit,NamespaceLifecycle,LimitRanger,ServiceAccount,ResourceQuota
+              {{ else }}
               - --admission-control=NamespaceLifecycle,LimitRanger,ServiceAccount,ResourceQuota
+              {{ endif }}
```
Comment (dghubble) [inline examples/ignition/k8s-master.yaml]: Also, I believe `endif` should be `end` in all these.

```
bootcfg[5]: time="2016-07-18T20:53:38Z" level=error msg="error parsing template: template: :165: function \"endif\" not defined"
```

Comment (mjg59) [inline examples/ignition/k8s-master.yaml]: Urgh yes sorry

**Comment by mjg59** [comment]
Rebased. I'll work on docs.

**Comment by dghubble** [comment]
Expecting to do this in this PR or another? We could have a dedicated `dtc.md` document under `Documentation`.

**Comment by mjg59** [comment]
I don't think baremetal is the place for it, other than just telling people how to enable it - the rest of the docs aren't going to depend on the deployment method

**Comment by dghubble** [comment]
I'll try out a deploy with this.

**Comment by dghubble** [comment]
This distributed trusted computing work was ahead of its time. Its not the focus at this moment, but I hope to revisit it someday.

**Pull request closed without merging**